### ISSUE TITLE
`tf.random.stateless_categorical` get different output when using same seed and same input

### ISSUE THREAD

**Issue by trfnhle** [tensorflow/tensorflow#73119]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.17.0

### Custom code

No

### OS platform and distribution

Ubuntu 22.04.3 LTS

### Python version

3.10.12

### CUDA/cuDNN version

12.2.140

### GPU model and memory

GPU T4

### Current behavior?

I run `tf.random.stateless_categorical` with the same inputs within a batch and the same seed but get different outputs. Is it suppose to get same output?

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
logits = tf.constant([[0.1, 0.2, 0.3, 0.4, 0.5, 0.6], [0.1, 0.2, 0.3, 0.4, 0.5, 0.6]], dtype=tf.float32)
sample_seed = tf.constant([3723, 3723], dtype=tf.int32)
output = tf.random.stateless_categorical(
        logits=logits,
        num_samples=1,
        seed=sample_seed,
        dtype=tf.int32,
    )
print(output)

```

### Relevant log output
```
tf.Tensor(
[[4]
 [2]], shape=(2, 1), dtype=int32)
```
